internal/pkg/bifs/system.go

```diff
@@ -44,3 +44,61 @@ func BIF_system(input1 *mlrval.Mlrval) *mlrval.Mlrval {
 
 	return mlrval.FromString(outputString)
 }
+
+func BIF_exec(mlrvals []*mlrval.Mlrval) *mlrval.Mlrval {
+
+	if len(mlrvals) == 0 {
+		return mlrval.ERROR
+	}
+
+	cmd := exec.Command(mlrvals[0].AcquireStringValue())
+	combinedOutput := false
+
+	args := []string { mlrvals[0].AcquireStringValue() }
+	if len(mlrvals) > 1 {
+		for _, val := range mlrvals[1].GetArray()[0:] {
+			args = append(args, val.AcquireStringValue())
+		}
+	}
+	cmd.Args = args
+
+	if len(mlrvals) > 2 {
+
+		for pe := mlrvals[2].AcquireMapValue().Head; pe != nil; pe = pe.Next {
+			if pe.Key == "env" {
+				env := []string {}
+				for _, val := range pe.Value.GetArray()[0:] {
+					env = append(env, val.AcquireStringValue())
+				}
+				cmd.Env = env
+			}
+			if pe.Key == "dir" {
+				cmd.Dir = pe.Value.AcquireStringValue()
+			}
+			if pe.Key == "combined_output" {
+				combinedOutput = pe.Value.AcquireBoolValue()
+			}
+
+			if pe.Key == "stdin_string" {
+				cmd.Stdin = strings.NewReader(pe.Value.AcquireStringValue())
+			}
+		}
+
```

Comment:
this could really use `if ... else if ... else` and handle the unexpected "none of the above" in the `else`

that said, miller DSL source code doesn't propagate go-style `error` returns as much as it should, so without reworking the `BIF` API, the best you could do would be to print to `stderr` and `exit 1` (which isn't great but is what i've done all over the place in the DSL)